shell/artifacts/Social/source/PostTimeRanker.js

```diff
@@ -0,0 +1,31 @@
+// @license
+// Copyright (c) 2018 Google Inc. All rights reserved.
+// This code may only be used under the BSD style license found at
+// http://polymer.github.io/LICENSE.txt
+// Code distributed by Google as part of this project is also
+// subject to an additional IP rights grant found at
+// http://polymer.github.io/PATENTS.txt
+
+'use strict';
+
+defineParticle(({Particle}) => {
+  return class PostTimeRanker extends Particle {
+    setViews(views) {
+      this.on(views, 'input', 'change', e => {
+        let inputHandle = views.get('input');
+        inputHandle.toList().then(input => {
+          // Rank the posts by creation time.
+          input = input.slice();
```

Comment:
Is it best practice to make a defensive copy like this or is it ok to directly mutate what we get back from `views.get('input')`? Since `Array.sort` modifies the input array.

Reply:
i don't think you need to copy, it should be ok to mutate the original array.

Reply:
Done.